commons-component-common/src/main/java/org/exoplatform/commons/persistence/impl/ExoTransactionalAspect.java

```
@@ -74,13 +77,15 @@ public Object around(ProceedingJoinPoint point) throws Throwable {
         throw e;
     } finally {
       // Do we need to end Transaction ?
-      if ((begunTx) && (entityManager.getTransaction().isActive())) {
+      if ((begunTx) && (tx.isActive())) {
         try {
           LOG.debug("Committing current transaction");
-          entityManager.getTransaction().commit();
+          tx.commit();
         } catch (RuntimeException e) {
-          LOG.error("Failed to commit current transaction. The transaction is rolled back.", e);
-          entityManager.getTransaction().rollback();
+          if (tx != null && tx.isActive()) {
+            LOG.error("Failed to commit current transaction. The transaction is rolled back.", e);
+            tx.rollback();
```

Ok, I'm going to fix the same potential leak in EntityManagerService.

ALSO, there is another concern that we should re-throw the exception of commit failure (or wrap it first) to higher level, instead of just logging ERROR for now. So they can know and handle the exception properly . For instance, UI could show the warning message about the action not done yet. WDYT ?